### PR TITLE
Correct the patches for HDF4 patches

### PR DESCRIPTION
**Pull request by varlackc**
This pull request fixes issues building the HDF4 package on graviton.

### CODE REVIEW

**Comment by pseudoyim** [comment]
@varlackc - can you please provide details on where the patches you added come from? (e.g. from conda-forge?) Thank you.

**Review by chenghlee — APPROVED** [review]
Echoing @pseudoyim's comments. We need to know where these patches came from and why that source was chosen. Other issues to address:
1. Is there an upstream release that actually works on AARCH64, or is this all (effectively) custom software development?
2. The patches need to be stripped down to cover just AARCH64. Currently, they seem to pull in all sorts of architectures (HPPA, M68K, etc.) that makes wrapping our heads around what the patches actually do very difficult.

**Review by chenghlee — CHANGES_REQUESTED** [review]
Echoing @pseudoyim's comments. We need to know where these patches came from and why that source was chosen. Other issues to address:

1. Is there an upstream release that actually works on AARCH64, or is this all (effectively) custom software development?
2. The patches need to be stripped down to cover just AARCH64. Currently, they seem to pull in all sorts of architectures (HPPA, M68K, etc.) that makes wrapping our heads around what the patches actually do very difficult.

**Comment by varlackc** [comment]
The patches came from Conda-Forge the link is provided below: 
https://github.com/conda-forge/hdf4-feedstock/tree/master/recipe/patches

The patches were selected because there were several types issues that were already solved with the conda-forge patches.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,10 +9,17 @@ source:
...
-        - max_files.patch
-
+        # - max_files.patch
+        - patches/max_files.patch
+        - patches/0001-Don-t-build-ctest0-and-run-ncgen-when-not-testing.patch
+        - patches/debian/64bit.patch
+        # - patches/debian/hdf-4.2.10-aarch64.patch
+        - patches/debian/hdf-4.2.10-aarch64.01.patch
+        - patches/debian/hdfi.h.01.patch
+        # - patches/debian/hdfi.h.patch
```
Comment (chenghlee) [inline recipe/meta.yaml:19]: Why do we have multiple copies of these patches? If we are going to adopt the conda-forge packaging & patches, we should be following the names within their recipe.

**Pull request closed without merging**